nengo_gui/components/spa_similarity.py

```diff
@@ -0,0 +1,102 @@
+from .pointer import Pointer
+from nengo_gui.components.component import Component
+
+import numpy as np
+import nengo
+
+import struct
+import copy
+
+class SpaSimilarity(Pointer):
+
+    config_defaults = dict(max_value=1,
+                           min_value=-1,
+                           show_pairs=False,
+                           **Component.config_defaults)
+
+    def __init__(self, obj, **kwargs):
+        super(SpaSimilarity, self).__init__(obj, **kwargs)
+        try:
+            target_key = kwargs['target']
+        except KeyError:
+            target_key = kwargs['args']
+
+        self.old_vocab_length = len(self.vocab_out.keys)
+        self.old_pairs_length = 0
+        self.labels = self.vocab_out.keys
+        self.previous_pairs = False
+
+    def gather_data(self, t, x):
+        vocab = self.vocab_out
+
+        # if there's been a change in the show_pairs
+        # Don't know how I feel about the timing of this
+        # maybe trigger on_message?
+        if self.config.show_pairs != self.previous_pairs:
+            #send the new labels
+            if self.config.show_pairs:
+                vocab.include_pairs = True
+                self.data.append(
+                    '["show_pairs_toggle", "%s"]' %(
+                        '","'.join(vocab.keys + vocab.key_pairs)))
+                # if we're starting to show pairs, track pair length
+                self.old_pairs_length = len(vocab.key_pairs)
+            else:
+                # Hmmm... an OH HELL NO was triggered...
+                vocab.include_pairs = False
+                self.data.append(
+                    '["show_pairs_toggle", "%s"]' %(
+                        '","'.join(vocab.keys)))
+
+        if(self.old_vocab_length != len(vocab.keys)):
+            # pass all the missing keys
+            legend_update = []
+            legend_update.append(vocab.keys[-1])
```

Comment:
This seems to assume that at most one new key is added at a time? Is this always True? It seems to me that multiple keys at once could be added.